### ISSUE TITLE
Why log metrics using the metric objects as keys in EvaluateAllMetrics?

### ISSUE THREAD

**Issue by matthew-z** [NTMC-Community/MatchZoo#790]
### Describe the Question
EvaluateAllMetrics updates logs using the metric objects as dict keys, which makes difficult to access logs. For example, `keras.callbacks.EarlyStopping` assumes string keys in logs....

**Comment by uduse** [comment]
matchzoo metrics have parameters, so only using class names won't work. Using generated names based on metric parameters work, but it's not very different from using objects.

**Comment by matthew-z** [comment]
Thank you for your kind explanation. I will use a modified version of `EvaluateAllMetrics` on my own to utilise `keras.callbacks.EarlyStopping`
